File: WebClinic/Models/Appointments.cs
using System.ComponentModel.DataAnnotations;

namespace WebClinic.Models
{
    public class Appointments
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public DateTime date { get; set; }

        public int pacientId  { get; set; }
        public int medicId { get; set; }
    }
}

File: WebClinicFrontend/src/app/services/appointments.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Appointments } from '../pages/appointments/appointments';

@Injectable({
  providedIn: 'root',
})
export class AppointmentsService {
  private apiUrl = "https://localhost:7132/api/appointments";

  constructor(private http: HttpClient){}

  getAppointments(){
   return this.http.get<any[]>(this.apiUrl);
  }

  getAppointmentsByPacientId(id: string){
   return this.http.get<any[]>(`${this.apiUrl}?pacientId=${id}`);
  }

  addAppointment(appointment: {date: string, pacientId: string, medicId: string}){
    return this.http.post<any>(this.apiUrl, appointment)
  }
  deleteAppointment(id: number){
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }
}
